### PR TITLE
Copying stuff from workbench/libproject to new utils repo. 

### PR DESCRIPTION
**Pull request by danqrodney**
Will be used in client repos as well.

### CODE REVIEW

**Comment by danqrodney** [comment]
(The thought being, I will add this repo as a submodule of workbench and client repos, replacing the libproject directory.)

**Comment by danqrodney** [comment]
I tried adding the git-secrets stuff. Let me know if this looks right.

File: hooks/apply-git-secrets.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+set -euo pipefail
+IFS=$'\n\t'
+
+{ which git-secrets; } || {
+  echo 'git-secrets required and not found. See README for installation instructions.'
```
Comment (s-rubenstein) [inline hooks/apply-git-secrets.sh]: The installation instructions need to be in the readme for this to be correct.

**Review by s-rubenstein — APPROVED** [review]
One comment on documentation, otherwise LGTM.

**Comment by danqrodney** [comment]
Reverting hooks addition for the time being.

**Pull request merged**